grails-app/controllers/org/pih/warehouse/api/LocalizationApiController.groovy

```diff
@@ -21,30 +21,42 @@ class LocalizationApiController {
     GrailsApplication grailsApplication
 
     def list() {
-        String languageCode = params.lang
+        String languageCode = params.languageCode
         String prefix = params.prefix
 
         String[] supportedLocales = grailsApplication.config.openboxes.locale.supportedLocales
 
+        Locale defaultLocale = Locale.default
         Locale currentLocale = localizationService.getCurrentLocale()
-        Locale defaultLocale = currentLocale.getDefault()
         Locale selectedLocale = languageCode ? LocalizationUtil.getLocale(languageCode) : currentLocale
 
-        // Get the default message properties as well as the message properties for the selected locale
+        // Get the system default message properties
         Properties defaultMessageProperties = localizationService.getMessagesProperties(defaultLocale)
+
+        // Get the country fallback language default message properties if it is country specific language
+        // (Example: if we have es_MX we want to have messages fallback in this precedence es_MX -> es -> en)
+        Properties fallbackMessageProperties = null
+        if (selectedLocale.country && supportedLocales.contains(selectedLocale.language)) {
+            Locale fallbackLocale = new Locale(selectedLocale.language)
+            fallbackMessageProperties = localizationService.getMessagesProperties(fallbackLocale)
+        }
+        // Get the message properties for the selected locale
         Properties selectedMessageProperties = localizationService.getMessagesProperties(selectedLocale)
 
         // Get all translations for the given prefix and locale from the database
-        List<Localization> localizedMessages = prefix ? Localization.findAllByCodeIlikeAndLocale("${prefix}%", selectedLocale.language) :
-                Localization.findAllByLocale(selectedLocale.language)
+        List<Localization> localizedMessages = prefix ? Localization.findAllByCodeIlikeAndLocale("${prefix}%", selectedLocale.toString()) :
+                Localization.findAllByLocale(selectedLocale.toString())
```

Comment:
For the record, these suggestions ^^ are for the longer term solution not for this PR. We should create a discovery or tech debt for any/all of the bullets. In the very near future we should definitely try to add a test or two to give us a sense of confidence that it will be working as expected.